src/DataTable/DataTable.js

```diff
@@ -414,10 +416,46 @@ const DataTable = memo(({
                         conditionalRowStyles={conditionalRowStyles}
                         selected={selected}
                         selectableRowsHighlight={selectableRowsHighlight}
+                        footer={false}
                         rowIndex={i}
                       />
                     );
                   })}
+
+                  {footer && (
+                    <TableRow
```

Comment:
These are an awful lot of props that will never be used for a footer row.

Reply:
Perhaps a footer should be its own component `<Footer />` similar to `subHeaderComponent` Footer can maybe have access to some table state similar to `expandableRowsComponent`

Reply:
For example, if aligning columns is not needed

https://github.com/jbetancur/react-data-table-component/blob/master/src/DataTable/ExpanderRow.js

this allows you to pass in `expandableRowsCompent={({ data }) => <div>{data.length}</div>}`

So let you can pass in whatever Footer you would like

`footerComponent={({ data, columns.... etc }) => <div>Count: {data.length}</div>}`

Reply:
I guess all rows data to do calculations... Columns should be aligned with the ability to do colspans. And expandable seems like a nice option. Too much? 🤯

Reply:
I think a footer should be sticky, or at least be given the option to be, similar behaviour to the header, where it can be incorporated in the flow of the rows but also have the option to be fixed at the top so you can see what the row values relate to.